### DISCUSSION TITLE
Get a click event of  a a-frame component

### DISCUSSION THREAD

**Discussion by pedrohmietlicki** [mozilla/hubs#3737] [General Questions & Discussion] [answered]
Hi, I want handle click event of dialog component like this 
![image](https://user-images.githubusercontent.com/33102027/105236363-d1c52d80-5b4a-11eb-94ef-2b0ace2e1eb4.png) and I try add EventListener('click',function) but don't worked.

**Comment by robinkwilson** [comment, marked as answer]
This unfortunately doesn't exist in the core product yet. We have an epic open for extensibility where interactive components are mentioned: https://github.com/mozilla/hubs/issues/1976

Definitely something we want to see in the future. There's a few folks that have deployed a custom client to get a click event, but it would require custom code.

**Reply by robinkwilson** [reply]
Here's another issue to follow: https://github.com/mozilla/hubs/issues/2635

**Comment by conath** [comment]
I found this discussion while looking to make a custom element clickable. I ended up with the following proof-of-concept A-Frame component (based on the [Spoke Rotating Cube example](https://github.com/mozilla/Spoke/blob/master/docs/creating-custom-elements.md)):

```javascript
AFRAME.registerComponent("rotate", {
  schema: {
    speed: { type: "number", default: 10 }
  },

  // test click
  init() {
    this.el.setAttribute("class", "interactable"); // This makes the object targetable by the cursor-targetting-system
    this.el.setAttribute("is-remote-hover-target", ""); // Required otherwise no interaction event (tested with mouse cursor)
    this.el.setAttribute("tags", { singleActionButton: true }); // Crucial
    this.onClick = () => {
      this.data.speed = this.data.speed > 300 ? 200 : 400;
    };
  },

  play() {
    this.el.object3D.addEventListener("interact", this.onClick);
  },

  pause() {
    this.el.object3D.removeEventListener("interact", this.onClick);
  },

  // rotating cube example
  tick(_time, dt) {
    const obj = this.el.object3D;
    obj.rotation.y += (dt / 1000) * THREE.Math.degToRad(this.data.speed);
    obj.matrixNeedsUpdate = true;
  }
});
```

**Reply by nhatdn** [reply]
i tried it but onClick function didn't work
